CRM/Volunteer/BAO/Assignment.php

```diff
@@ -247,10 +247,12 @@ private static function setActivityDefaults(array $params) {
       $defaults['campaign_id'] = '';
     }
 
+    // Taking volunteer_role_id outside of Action ADD scope, we have to update the role on moving the activity.
+    $defaults['volunteer_role_id'] = CRM_Utils_Array::value('role_id', $need);
     if ($op === CRM_Core_Action::ADD) {
-      $defaults['volunteer_role_id'] = CRM_Utils_Array::value('role_id', $need);
       $defaults['time_scheduled_minutes'] = CRM_Utils_Array::value('duration', $need);
-      $defaults['target_contact_id'] = CRM_Volunteer_BAO_Project::getContactsByRelationship($project->id, 'volunteer_beneficiary');
+      $defaults['target_contact_id'] = CRM_Volunteer_BAO_Project::getContactsByRelationship($project->id, 'volunteer_manager'); // Changed to volunteer_manager from volunteer_beneficiary.
```

Comment:
@agileware, this change seems wrong to me. The target of the activity should be the beneficiary. Ultimately we want the activity to indicate that Joe Volunteer (activity assignee) was assigned by whomever (usually himself, but sometimes a staff member, if added in the backend -- anyway, the activity source or "added by" contact) to volunteer for Acme Charity (the activity target, aka the "with" contact).

A number of chaptered organizations are using CiviVolunteer, and this field allows them to distinguish between volunteers for Acme Charity New York vs. Acme Charity San Francisco.